bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```
@@ -0,0 +1,449 @@
+            config = getConfigAs(MillheatAccountConfiguration.class);
+            LoginResponse rsp = sendLoginRequest(new LoginRequest(config.username, config.password),
+                    LoginResponse.class);
+            int errorCode = rsp.errorCode;
+            if (0 != errorCode) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        String.format("Error login in: code=%s, type=%s, message=%s", errorCode, rsp.errorName,
+                                rsp.errorDescription));
+            } else {
+                // No error provided on login, proceed to find token and userid
+                token = StringUtils.trimToNull(rsp.token);
+                userId = rsp.userId == null ? null : rsp.userId.toString();
+                if (token == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "error login in, no token provided");
+                } else if (userId == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "error login in, no userId provided");
+                } else {
+                    return true;
+                }
+            }
+        } catch (MillheatCommunicationException e) {
+            logger.info("Error login", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error login: " + e.getMessage());
+        }
+        return false;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(MillheatAccountConfiguration.class);
+        scheduler.execute(() -> {
+            if (doLogin()) {
+                try {
+                    model = refreshModel();
+                    updateStatus(ThingStatus.ONLINE);
+                    discoveryService.startService();
+                    initPolling();
+                } catch (Exception e) {
```

Do you need to catch all exceptions here, or can you handle them more specific?